infra/modules/notifications/resources/variables.tf

```diff
@@ -0,0 +1,20 @@
+variable "domain_name" {
+  description = "The domain name to configure SES."
+  type        = string
+}
+
+variable "name" {
+  type        = string
+  description = "Name of the notifications project/application"
+}
```

Comment:
Get rid of one of these variables to keep the interface simple and also so that we can have the interface to the data module be a single variable. I lean towards getting rid of `name`. `domain_name` is more explicit which I think is good. Resources that rely on `name` for naming thing like the logs resource can use `domain_name` instead.